tests/utils/streams/test_batching.py

```diff
@@ -1,31 +1,66 @@
 import time
 from datetime import datetime
-from typing import Any, MutableSequence, Sequence, Optional
+from typing import Any, Callable, Mapping, MutableMapping, MutableSequence, Sequence, Optional
 from unittest.mock import patch
 
-from confluent_kafka import Message
-
 from snuba.utils.metrics.backends.dummy import DummyMetricsBackend
+from snuba.utils.streams.abstract import Consumer
 from snuba.utils.streams.batching import AbstractBatchWorker, BatchingKafkaConsumer
-from tests.backends.confluent_kafka import FakeConfluentKafkaConsumer, FakeConfluentKafkaMessage, FakeConfluentKafkaProducer
+from snuba.utils.streams.kafka import KafkaMessage, TopicPartition
+from tests.backends.confluent_kafka import FakeConfluentKafkaProducer
+
+
+class FakeKafkaConsumer(Consumer[TopicPartition, int, bytes]):
+    def __init__(self):
+        self.items: MutableSequence[KafkaMessage] = []
+        self.commit_calls = 0
+        self.close_calls = 0
+        self.positions: MutableMapping[TopicPartition, int] = {}
+
+    def subscribe(
+        self,
+        topics: Sequence[str],
+        on_assign: Optional[Callable[[Sequence[TopicPartition]], None]] = None,
+        on_revoke: Optional[Callable[[Sequence[TopicPartition]], None]] = None,
+    ) -> None:
+        pass  # XXX: This is a bit of a smell.
```

Comment:
I think this is more of a sign of a test that doesn't fully encapsulate the state space, rather than a sign of a bad abstraction. There are cases where subscription might not be necessary (e.g. "consuming" from a file to load a table) so maybe I'm wrong.

Reply:
> There are cases where subscription might not be necessary (e.g. "consuming" from a file to load a table) so maybe I'm wrong.

We could move fixtures for development and testing into a directory, where "topics" map to file or directory structures, and `TStream` is `io.BytesIO` object (or something representing one) and `TOffset` would be the stream position (for `seek`/`tell`.)